lib/heap_sort.rb

```diff
@@ -1,8 +1,19 @@
-
+require_relative './min_heap'
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity:  o(n log(n))
+# Space Complexity: o(n)
+def heapsort(list)
```

Comment:
One showstopper of a bug preventing this from working.  See my suggestion below.